frontend/src/metabase/components/Header.jsx

```diff
@@ -1,28 +1,71 @@
+/* @flow */
 import React, { Component } from "react";
-import ReactDOM from "react-dom";
+import { findDOMNode } from "react-dom";
 
-import Input from "metabase/components/Input.jsx";
-import HeaderModal from "metabase/components/HeaderModal.jsx";
-import TitleAndDescription from "metabase/components/TitleAndDescription.jsx";
-import EditBar from "metabase/components/EditBar.jsx";
+import Input from "metabase/components/Input";
+import HeaderModal from "metabase/components/HeaderModal";
+import TitleAndDescription from "metabase/components/TitleAndDescription";
+import EditBar from "metabase/components/EditBar";
 
 import { getScrollY } from "metabase/lib/dom";
 
+type Props = {
+    // Buttons to be displayed in the header in default mode
+    defaultActions: [],
```

Comment:
@attekei When you get a chance I'd dig your feedback on how I've refactored this `Header` component. I wasn't a huge fan of all the if else stuff in `DashboardHeader` to change what was in `headerButtons` based different conditions. Instead I figured, let's just have explicit ways to add buttons that need to be present in the default, editing, and fullscreen states.

Reply:
Yeah, in general makes lots of sense! `Header` encapsulates now pretty complicated behavior so it takes a really high number of props... but it's not a big concern to me. 

Is your idea that this would be applied to places like QB and data reference page too? Is the idea to bring more consistency in terms of header design? Now dashboards seem to be only place in the app where we have this combination of editable title/description header AND a separate "You are editing entity X" bar above that header.